### ISSUE TITLE
Keyboard → Switch Layout shortcut conflicts with Alt+Tab

### ISSUE THREAD

**Issue by cassidyjames** [elementary/default-settings#4]
Our default "Switch layout" shortcut for switching between keyboard layouts is Alt+Shift, but that conflicts with and breaks Shift+Alt+Tab to switch windows backwards. We should disable or change this shortcut since we keep getting bugs about Shift+Alt+Tab being "broken".

Moved from https://github.com/elementary/gala/issues/18

**Comment by snowparrot** [comment]
I really recommend to change the shortcut as working in different languages gets much more convenient with a shortcut.

**Comment by Philosoft** [comment]
This is a big problem indeed, but "change one of conflicting shortcuts" is not an option - it's just avoiding and postproning general problem. If `change layout` is bind to <kbd>alt</kbd>+<kbd>shift</kbd>, then it brokes every other shortcut wich starts with <kbd>alt</kbd>+<kbd>shift</kbd>, if you change it to something else, some other shortcuts will be broken. why windows, mac, gnome, kde etc can solve this problem and elementery can't?

Also on topic.  <kbd>alt</kbd>+<kbd>shift</kbd>, <kbd>alt</kbd>+<kbd>tab</kbd>,  <kbd>alt</kbd>+<kbd>shift</kbd>+<kbd>tab</kbd> is common across desktops and countries, why change defaults?

Most of all, why can't I map my own shortcut to switch layouts?

**Comment by lewisgoddard** [comment]
I assume this a problem somewhere where the shortcuts are detected and passed off. Maybe we are detecting `alt`+`shift` on keydown instead of keyup?

**Comment by davidmhewitt** [comment]
This is a very long standing issue in X itself, GNOME is almost certainly affected too.


I think our only option really is to bind the language switching shortcut to something that doesn't conflict. Which is easier said than done, since you could move it to Ctrl+Shift, but then Ctrl+Shift+Tab would stop working for switching tabs in a web browser. But a slightly less common conflict than Alt+Shift+Tab I would guess

**Comment by lewisgoddard** [comment]
Why not something that doesn't conflict at all, like Ctrl+Alt+L?

**Comment by Philosoft** [comment]
@lewisgoddard i, myself, would prefer <kbd>AltGr</kbd> to do so, but there are not any suitable options (or as you propose custom <kbd>ctrl</kbd>+<kbd>alt</kbd>+<kbd>L</kbd>. screenshot:

![screenshot from 2018-02-14 10 32 41](https://user-images.githubusercontent.com/296326/36192423-7f26cba8-1172-11e8-9007-c39cd25e8423.png)

as I additionaly notice, you just cannot screenshot popup menu with <kbd>PrtSc</kbd> - it simply doesn't work.

**Comment by danrabbit** [comment]
I'm going to bump this to beta2 since it doesn't imply any string or UI changes

**Comment by danrabbit** [comment]
I'm going to drop this from beta2 since it seems discussion is still ongoing and an agreement hasn't yet been reached

**Comment by peteruithoven** [comment]
How did other Distro's fix this?

@danrabbit @cassidyjames Maybe one of you (being the most public) could do a poll on Twitter and/or Reddit?

**Comment by cassidyjames** [comment]
Windows uses Alt+Shift, but it seems problematic to bind it to only modifier keys that are used in other shortcuts out of the box. GNOME uses Super+Space which we use for Applications Menu. It looks like macOS uses Ctrl+Space/Alt+Ctrl+Space to cycle in different directions: https://support.apple.com/guide/mac-help/type-language-mac-input-sources-mchlp1406/mac

I'm going to propose we change the default key in elementary OS to not conflict with the Alt+Tab switcher. Alt+Space seems like a sane option, or if possible, Ctrl+Space. 

And remember, users can always change it to something that is more familiar/comfortable for them in System Settings → Keyboard.

**Comment by tom-james-watson** [comment]
Sounds like a sane change. I think Alt+Space is a good choice, personally.

**Comment by janxkoci** [comment]
I agree with **alt+space**, as ctrl+space is very popular with 3rd party launchers like Synapse, Kupfer, GNOME Do, and others (and I think some don't allow to change the combo).

**Comment by tom-james-watson** [comment]
I've not tried it but from [some blog posts](https://www.omgubuntu.co.uk/2021/11/elementary-os-6-intros-a-simpler-alt-tab-experience) I read, it sounds like this issue has been fixed and that alt+shift+tab should now work?

**Comment by cmonagle** [comment]
Is there an update on this? It looks like @cassidyjames was proposing Alt+Space. Would a PR be welcome to implement this? As far as I can tell it should be a few lines in this repo? If that's the case I'm happy to open one (but if it's much more than that I'm afraid I may need help or to pass it off).

**Comment by o-alquimista** [comment]
<kbd>Super</kbd> + <kbd>Space</kbd> already works by default, doesn't it? The app menu is now opened with a single <kbd>Super</kbd> key-press.
